pkgs/top-level/all-packages.nix

```diff
@@ -26597,6 +26606,26 @@ with pkgs;
 
   thttpd = callPackage ../servers/http/thttpd { };
 
+  stalwart-mail_0_6 = stalwart-mail.overrideAttrs (old: rec {
+    pname = "stalwart-mail_0_6";
+    version = "0.6.0";
+    src = fetchFromGitHub {
+      owner = "stalwartlabs";
+      repo = "mail-server";
+      rev = "v${version}";
+      hash = "sha256-OHwUWSUW6ovLQTxnuUrolQGhxbhp4YqKSH+ZTpe2WXc=";
+      fetchSubmodules = true;
+    };
+    cargoDeps = old.cargoDeps.overrideAttrs (_: {
+      inherit src;
+      outputHash = "sha256-M4mAUI4hBso9rlSrRxpjO/DoWqKwg38kMAhb0E33onE=";
+    });
+    env = old.env // {
```

Comment:
Instead of tinkering with the `env` here, we could also just use `override` to use the desired rocksdb.
Due to the planned migration to by-name here we sadly have to do `(stalwart-mail.override { rocksdb_8_11 = rocksdb_8_3; })`
which might be a bit counter intuitive. 
If we would not migrate to by-name, we could use `(stalwart-mail.override { rocksdb = rocksdb_8_3; })` and use ` stalwart_mail = (callPackage ../servers/mail/stalwart { }).override { rocksdb = rocksdb_8_11; }`  for the "new" version until the rust crate catches up with the rocksdb version.
Not  saying that its optimal, i am also totally fine keeping the code like it is.